### DISCUSSION TITLE
PHp 8.3

### DISCUSSION THREAD

**Discussion by tbanov** [stackkit/laravel-google-cloud-scheduler#27] [General] [answered]
Hello, thank you for your great work

What is the current status of php 8.3 support

Best, Tiho

**Comment by marickvantuil** [comment, marked as answer]
> Hello, thank you for your great work

Thank you!

`v2.0.5` should work with PHP 8.3. Let me know if you run into issues.

**Reply by tbanov** [reply]
Thank you for your assistance. It appears there may be an issue on my end or with Cloud Run. The schedule:run command does not seem to be executing as expected. While changing it to a specific command works, I'm unable to redistribute the scheduler. I'll keep you updated once I've resolved the issue.

**Reply by tbanov** [reply]
For the generations ahead :)

Here is the fix

```
 Don't work
 
 $schedule->command(SchedulerTest::class)->everyMinute();
```

```
 Works
 
 $schedule->call(function () {
      app(SchedulerTest::class)->handle();
 })->everyMinute();
 ```

**Reply by marickvantuil** [reply]
Hi, I can't reproduce this on Laravel 10.47.0 and PHP 8.3. Do you mind sharing some more info about your setup (versions, what the command is doing)? Then we can hopefully find what's going wrong.

**Reply by marickvantuil** [reply]
@tbanov I've possibly found why it might not be working. I was able to reproduce it as well. In my case, the path to the PHP executable was empty:

```php
(new Symfony\Component\Process\PhpExecutableFinder())->find(); // false
``` 

After digging into the PhpExecutableFinder, it got stuck on this line:

```php
if (@is_executable($php = \PHP_BINDIR.('\\' === \DIRECTORY_SEPARATOR ? '\\php.exe' : '/php')) && !@is_dir($php)) {
    return $php;
}
```

After removing the `@` error suppression, the error was revealed:

```log
[2024-03-15 21:51:01] local.ERROR: is_executable(): open_basedir restriction in effect. File(/usr/bin/php) is not within the allowed path(s): (/var/www/html:/dev/stdout:/tmp) {"exception":"[object] (ErrorException(code: 0): is_executable(): open_basedir restriction in effect. File(/usr/bin/php) is not within the allowed path(s): (/var/www/html:/dev/stdout:/tmp) at /var/www/html/vendor/symfony/process/PhpExecutableFinder.php:75)
[stacktrace]
#0 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Bootstrap/HandleExceptions.php(255): Illuminate\\Foundation\\Bootstrap\\HandleExceptions->handleError()
#1 [internal function]: Illuminate\\Foundation\\Bootstrap\\HandleExceptions->Illuminate\\Foundation\\Bootstrap\\{closure}()
#2 /var/www/html/vendor/symfony/process/PhpExecutableFinder.php(75): is_executable()
```

After adding `/usr/bin/php` to the open_basedir PHP setting, it works again.
